lib/ansible/executor/task_executor.py

```diff
@@ -840,7 +840,12 @@ def _poll_async_result(self, result, templar, task_vars=None):
         # that (with a sleep for "poll" seconds between each retry) until the
         # async time limit is exceeded.
 
-        async_task = Task.load(dict(action='async_status', args={'jid': async_jid}, environment=self._task.environment))
+        async_task = Task.load(dict(
+            action='async_status',
+            args={'jid': async_jid},
+            check_mode=self._task.check_mode,
```

Comment:
The clean up task below https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/task_executor.py#L909-L917 needs this too.

Reply:
Thanks for the pickup, I must have missed it because we don't check the result there.